From 862ec08536a0aa67b2a841b23ebdbeb5beaf67da Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 24 Sep 2026 17:01:34 +0800
Subject: [PATCH] FROMLIST: Set LG_QUANTUM to 4 for SuperH-3

The current code does not define LG_QUANTUM for SH-3, only SH-3E.

Build tested on a SuperH-3 baseline sysroot.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Link: https://github.com/jemalloc/jemalloc/pull/3013

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 include/jemalloc/internal/quantum.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/jemalloc/internal/quantum.h b/include/jemalloc/internal/quantum.h
index 2f7c0466..8e0af5cc 100644
--- a/include/jemalloc/internal/quantum.h
+++ b/include/jemalloc/internal/quantum.h
@@ -60,7 +60,8 @@
 #	ifdef __s390__
 #		define LG_QUANTUM 4
 #	endif
-#	if (defined(__SH3E__) || defined(__SH4_SINGLE__) || defined(__SH4__)  \
+#	if (defined(__SH3__) || defined(__SH3E__) \
+	    || defined(__SH4_SINGLE__) || defined(__SH4__) \
 	    || defined(__SH4_SINGLE_ONLY__))
 #		define LG_QUANTUM 4
 #	endif
-- 
2.55.0

